### ISSUE TITLE
Unable to zoom on cover image

### ISSUE THREAD

**Issue by nerdboy4200** [koreader/koreader#6409]
* KOReader version: 2020.07 and possibly 2020.06 as well
* Device: Kobo Aura One

#### Issue
Opening several epub books - unable to zoom the cover image

#### Steps to reproduce
* open epub book
* hold finger down on cover image - in a second or 2 it should zoom to the whole screen

#### Notes
* This also appears to affect an old books as well (for example a book that had to have its metadata upgraded)
* Testing the same books on the linux appimage shows no issue

**Comment by poire-z** [comment]
> Opening several epub books - unable to zoom the cover image

Can't reproduce: I opened 2 books on my Kobo Glo HD, got back to start of book, long press on the image there (that is the cover), and it shows in the ImageViewer.
Do you have that issue only on the cover image on the 1st page, or also in smaller images later in the books (if you have books with images).

Or may be these books are twisted, and the cover image is not a classic `<IMG SRC="...">` - but somehow set via CSS `background-image: url(...)` ?

**Comment by NiLuJe** [comment]
No issues on my end either, FWIW.

**Comment by nerdboy4200** [comment]
Checked the source - the title page shows the cover code as this:

```
<image width="501" height="751" xlink:href="cover.jpeg"/>
```

Another Kobo that we have tested the same books on works properly.

Now in one of the books - there are image separators in the chapters that are these:
```
<img src="../images/00004.jpeg" alt="image" class="calibre3"/>
```

and they can't be brought up easier.

Now the this makes me think of 2 possible cases:
* the holding code is broken - which could explain why it doesn't trigger - specifically with the Kobo Aura One.
* broken imageviewer that fails when it starts - so it does nothing

I'm trying to debug this via -d option - but at this point - I have only been able to do this on my system with the linux appimage.

I'm trying to determine how to run koreader with -d over ssh on the Kobo - but it appears that ksm keeps control of the touch inputs when its running - so I'm not sure how to make it work properly.

**Comment by poire-z** [comment]
> Another Kobo that we have tested the same books on works properly.
> the holding code is broken - which could explain why it doesn't trigger 

So, might be your settings. Does long-press on text works and do dict lookup?
Backup/rename your settings.reader.lua - so you start with default settings, just to see if that's that.

> I'm trying to debug this via -d option

Or just replace the `logger.dbg` with `logger.info` (and add a few more) in this file on your Kobo, to get a quick idea of what's (not) happening:
https://github.com/koreader/koreader/blob/6e7c7bb8416ceb99abefce68027477288d2f6598/frontend/apps/reader/modules/readerhighlight.lua#L576-L609

**Comment by nerdboy4200** [comment]
OK - so there appears to be 2 things here:
1) figured it out - somehow the default of enabling highlighting got disabled (took a bit to understand that you had to hold it down to set the default)

It turns out that it worked after doing an upgrade of an old books metadata - since at the time - the highlight option was enabled. If the .sdr cache was cleared - it magically stopped working. This happened because the default of enable highlight got disabled somehow.

2) I think that the highlight option might be incorrect for this sort of feature.

I'm not sure that image zoom in a book is really considered a highlight or not. I'd think it would be independent of highlighting.

**Comment by poire-z** [comment]
> I'm not sure that image zoom in a book is really considered a highlight or not. I'd think it would be independent of highlighting.

I would agree. I had a look, and thought it would just be a matter of moving `if self.view.highlight.disabled then return false end` after the check for image, and it may.
But I'm not sure what this `[ ] Allow highlighting` might be used for.
Currently, it's more of a "ignore long-press on book content", which might be useful - it prevents anything long-press: zooming on image, single word dict lookup (which is not highlighting)...
Just disabling highlighting feels like a limited/stupid use case, or does it not?

There are 2 menu places that allows tweaking highlighting:
- `Typeset icon > Highlighting> [ ] Allow + type` (lighten, underline, invert, which indeed feels a bit like typesetting, at least how highlights will look :)
- `Gear icon > Document > Highlight action`, which is more about "Long-press" action. It's a bit mixed as it's more about what happens after the extension of a single word to multiple words, but it has a `[ ] Enable on single word selection`. If we add to it a `[ ] Do nothing`, we could have a "disable long-press" setting. And then, may be `[ ] Allow highlighting`  could have more sense - or we can just remove it if not.

**Comment by poire-z** [comment]
Any thought on the above ? (I guess nobody had any if no answer... but I insist: please have a thought! :)

**Comment by Frenzie** [comment]
The image zoom only works when highlighting is enabled?

**Comment by poire-z** [comment]
Yep.

**Comment by Frenzie** [comment]
How do you disable highlighting?

**Comment by poire-z** [comment]
In the first menu place mentionned above in https://github.com/koreader/koreader/issues/6409#issuecomment-776682190 : 
> `Typeset icon > Highlighting> [ ] Allow `

**Comment by Frenzie** [comment]
I think it makes the most sense to just tweak the text a little rather than the behavior.

**Comment by poire-z** [comment]
Pinging @hius07 about https://github.com/koreader/koreader/issues/6409#issuecomment-776682190 just in case - as you're not afraid having a look at and putting your mind on reworking/reorganising all these little settings we usually don't care much about :)

**Comment by hius07** [comment]
Thanks for pointing to, I'll think about it.

**Comment by hius07** [comment]
Summarizing once more the comprehensive https://github.com/koreader/koreader/issues/6409#issuecomment-776682190:

(1) three long-press actions (highlight text, single word dictionary lookup, image view) are rather independent and should be enabled/disabled with separate settings.

The question: does anybody need to disable dictionary lookup or image view?

If yes (to disable long-press in full)
(2) we may have these separate settings in the `Gear - Document - Long-press actions` menu (common settings)

If not (dictionary lookup and image view will always be enabled on long-press)
(3) as said, we may have an option `Gear - Document - Long-press actions - Do nothing` to disable highlighting only
(do not forget to handle a special case of long-pressing longer than 3 s)

(4) The question: is it necessary to have a per document setting of enabling/disabling highlighting?
I'd say no. If a user generally disabled highlighting but wants to make notes in the current document, he can enable it while reading this book, make highlights and then disable back if needed. Obviously, the highlights remain in the book.
If so, we may have a per document `Typeset - Highlight style` menu only.

**Comment by poire-z** [comment]
No real idea about what would be the right menu items to make all that clear and consistent.
So, just answering the questions :)

> (1) does anybody need to disable dictionary lookup or image view?

Probably not.
Image are rares, so a long-press on them too.
No need to have a dedicated setting for disabling long-press on a single word, but if a combination of other settings allow that, that's good for the rare people who may want that (like, they would too often long-press while manipulating their device, cf https://github.com/koreader/koreader/pull/6998#issuecomment-748527179).
We can currently have that with `Highlight action> Do nothing` and `Enable on single word selection` I think. We don't need something more explicite for that rare use case.

> `Gear - Document - Long-press actions` 

Fine with having whatever can help in there :)
(Minor question: but then, even if it's related to long-press on a document page, shouldn't it be in `Tap & gestures>` ? `Page turns` and `Scrolling`, which are document only, are in there.)

> (4) The question: is it necessary to have a per document setting of enabling/disabling highlighting?

Nope, not necessary. I guess it's a user-generic-behaviour choice, that wouldn't change per document.

But again, re-reading my own comment above:
> Just disabling highlighting feels like a limited/stupid use case, or does it not?

Don't you think so ?
Long-press while moving the finger to select more word has even less risk of being accidental. So, if you never highlight, you would just change the action to be `Wikipedia lookup` - or `Do nothing`.
So, I think `Disable highlights` should be removed (because if allowed, we should still allow selecting multiple word to do other things like Translation, and disable "Highlight" in the dialog, which is mess of coding).
